### PR TITLE
sveltekit viewer document

### LINKED ISSUE #534: Document view
Render the pages of a document as in the original PDF.

### PR DESCRIPTION
**Pull request by eyeseast**
Closes #534 

- [x] render PDF
- [x] selectable text
- [x] zoom
- [x] pagination
- [x] loading optimization

### CODE REVIEW

**Review by mitchelljkotler — COMMENTED** [review]
This looks good overall.  I think it might be worth going over together at a tech only project planning meeting - it is a little hard to follow some of the PDFjs stuff going on.

**Review by allanlasser — CHANGES_REQUESTED** [review]
Amazing how this all came together. There's a few overflow issues when browsing at a width > fit view, I'm working on some style debugging for those.

File: src/routes/documents/[id]-[slug]/+layout.svelte [hunk 1]
```diff
@@ -3,6 +3,13 @@
 
   import type { Project } from "$lib/api/types";
 
+  // load this here to get the worker started early
+  import * as pdfjs from "pdfjs-dist/build/pdf.mjs";
+  pdfjs.GlobalWorkerOptions.workerSrc = new URL(
+    "pdfjs-dist/build/pdf.worker.mjs",
+    import.meta.url,
```
[the remark this answers is not in the document]
Comment (eyeseast) [reply src/routes/documents/[id]-[slug]/+layout.svelte:10]: No. The issue is [`Promise.withResolvers`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers). It's supported in the browser but not in Node until 22.

File: src/routes/documents/[id]-[slug]/+page.svelte [hunk 2]
```diff
@@ -167,10 +223,10 @@
           on:goTo={(e) => scrollToPage(e.detail)}
```
[the remark this answers is not in the document]
Comment (eyeseast) [reply src/routes/documents/[id]-[slug]/+page.svelte]: Done.

File: src/lib/utils/scroll.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ * Scroll to a page using a standard ID
+ *
+ * @param n page number
+ */
+export function scrollToPage(n: number): void {
+  const pageId = pageHashUrl(n).replace("#", "");
+  const heading = window.document.getElementById(pageId);
+
+  if (!heading) return console.error(`Missing page ${n}`);
+  heading.scrollIntoView();
```
[the remark this answers is not in the document]
Comment (eyeseast) [reply src/lib/utils/scroll.ts:13]: Added, and learned lots about DOM mocking.

File: src/lib/components/documents/PDF.svelte [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  {/each}
+</div>
+
+<style>
+  .pages {
+    display: flex;
+    flex-direction: column;
+    margin: 0 auto;
+    gap: 3rem;
+    width: 100%;
```
Comment (allanlasser) [inline src/lib/components/documents/PDF.svelte:69]: When the selected width overflows the container (e.g. 200%), this will make the overflow scrollable:

```suggestion
    width: 100%;
    overflow-x: auto;
```

File: src/lib/components/documents/Page.svelte [hunk 5]
```diff
@@ -1,39 +1,110 @@
...
     max-width: 100%;
   }
 
+  .page.wide {
+    width: 100%;
+  }
+
   h4,
   h4 a {
     color: var(--gray-4, #5c717c);
```
Comment (allanlasser) [inline src/lib/components/documents/Page.svelte:110]: Combined with the suggestion to have pages overflow horizontally when their width exceeds the container, this will keep the page indicator in place as the user scrolls around:

```suggestion
    position: sticky;
    left: 1rem;
    color: var(--gray-4, #5c717c);
```
